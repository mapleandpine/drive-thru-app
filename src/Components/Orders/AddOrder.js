import React, { Component } from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid';

class AddOrder extends Component {

  constructor(props) {
    super(props);
    this.state = {
      newOrder:{}
    }
  }

  static defaultProps = {
    categories: ['Cash', 'Credit', 'Gift Certificate']
  }

  handleSubmit(e) {
    if(this.refs.title.value === "") {
      alert('Title is required');
    } else {
      this.setState({newOrder:{
        id: uuid.v4(),
        title: this.refs.title.value,
        category: this.refs.category.value
      }}, function() {
        this.props.addOrder(this.state.newOrder);
      })
    }
    e.preventDefault();
  }

  render() {
    let categoryOptions = this.props.categories.map(category => {
      return <option key={category} value={category}>{category}</option>
    })
    return (
      <div className="AddOrder">
        <h3>Add Order</h3>
        <form onSubmit={this.handleSubmit.bind(this)}>
          <div>
            <label>Title</label><br />
            <input type="text" ref="title" />
          </div>
          <br />
          <div>
            <label>Category</label><br />
            <select ref="category">
              {categoryOptions}
            </select>
          </div>
          <br />
          <input type="submit" value="Submit"/>
          <br />
        </form>
      </div>
    );
  }
}

AddOrder.propTypes = {
  categories: PropTypes.array,
  addOrder: PropTypes.func
}

export default AddOrder;
