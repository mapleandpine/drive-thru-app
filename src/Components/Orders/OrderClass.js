import { Component } from 'react';
import uuid from 'uuid';

class OrderClass extends Component {

  // use this Order definition via `let myOrder = new Order()`
  // data definition and state should live top-level
  constructor() {
    super();
    this.closed = false;
    this.date = new Date();
    this.id = uuid.v4();
    this.items = [];
  }
}

export default OrderClass;
