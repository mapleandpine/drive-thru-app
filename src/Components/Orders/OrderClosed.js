import React, { Component } from 'react';
import PropTypes from 'prop-types';

class OrderClosed extends Component {

  activeOrder(id) {
    this.props.onActiveOrder(id);
  }

  deleteOrder(id) {
    this.props.onDeleteOrder(id);
  }

  render() {
    return (
      <div className="Order">
        <span className="OrderName" onClick={this.activeOrder.bind(this, this.props.order.id)}>{this.props.order.date.toLocaleTimeString()}</span>
        <button className="OrderDelete" onClick={this.deleteOrder.bind(this, this.props.order.id)}><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"/></svg></button>
      </div>
    );
  }
}

OrderClosed.propTypes = {
  onActiveOrder: PropTypes.func,
  onDeleteOrder: PropTypes.func
}

export default OrderClosed;
