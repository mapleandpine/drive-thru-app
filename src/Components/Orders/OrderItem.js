import React, { Component } from 'react';
import PropTypes from 'prop-types';

class OrderItem extends Component {

  deleteOrder(id) {
    this.props.onDelete(id);
  }

  render() {
    return (
      <li className="OrderItem">
        <strong>{this.props.order.title}:</strong> {this.props.order.category} <a href="#;" onClick={this.deleteOrder.bind(this, this.props.order.id)}>X</a>
      </li>
    );
  }
}

OrderItem.propTypes = {
  order: PropTypes.object,
  onDelete: PropTypes.func
}

export default OrderItem;
