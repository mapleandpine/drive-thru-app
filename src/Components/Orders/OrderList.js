import React, { Component } from 'react';
import PropTypes from 'prop-types';
import OrderListItem from "./OrderListItem";

class OrderList extends Component {

  deleteOrderList(id) {
    this.props.onDelete(id);
  }

  render() {
    let orderListItems;
    if(this.props.orderList) {
      orderListItems = this.props.orderList.map(menuItem => {
        return (
          <OrderListItem onDelete={this.deleteOrderList.bind(this)} key={menuItem.id} menuItem={menuItem} />
        )
      });
    }
    return (
      <div className="OrderList">
        {orderListItems}
      </div>
    );
  }
}

OrderList.propTypes = {
  orders: PropTypes.array,
  onDelete: PropTypes.func
}

export default OrderList;
