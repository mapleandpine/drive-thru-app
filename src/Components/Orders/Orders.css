/* ORDERS */

.Orders {
  display: flex;
  flex-direction: column;
  grid-column: col 2;
  grid-row: row;
  max-height: 300px;
  overflow: scroll;
  padding: 10px;
}

.Orders .heading {
  color: #555;
  text-transform: uppercase;
}

.Order {
  cursor: pointer;
  display: flex;
  margin-bottom: 10px;
}

.OrderName {
  animation: pulse .3s 2 ease-out;
  background-color: #EFCE5D;
  border-radius: 4px;
  color: #333;
  display: flex;
  flex-grow: 2;
  justify-content: space-between;
  margin-right: 5px;
  padding: 10px;
}

.OrderName:hover {
  background-color: #E2BE57;
}

.OrderComplete,
.OrderDelete {
  border-radius: 4px;
  color: #f5f5f5;
  display: inline-block;
  font-size: 36px;
  line-height: 24px;
  padding: 10px;
  text-align: center;
  width: 44px;
}

.OrderComplete svg,
.OrderDelete svg {
  fill: #fff;
}

.OrderComplete {
  background-color: #5FA866;
  margin-right: 5px;
}

.OrderComplete:hover {
  background-color: #4C8354;
}

.OrderComplete svg {
  fill: #fff;
}

.OrderDelete {
  background-color: #CE4E4E;
}

.OrderDelete:hover {
  background-color: #B2474D;
}
