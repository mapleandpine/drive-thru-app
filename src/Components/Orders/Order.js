import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Orders.css';

class Order extends Component {

  newOrder(e) {
    console.log('new order');
    // this.setState({newOrder: new Order()}, function() {
    //   this.props.addOrder(this.state.newOrder);
    // })
    e.preventDefault();
  }

  deleteOrder(id) {
    this.props.onDeleteOrder(id);
  }

  render() {
    return (
      <li className="Order">
        <span className="OrderName">{this.props.order.date.toLocaleTimeString()}</span> <a className="OrderNew" onClick={this.newOrder.bind(this)}>New</a> <a className="OrderDelete" href="#;" onClick={this.deleteOrder.bind(this, this.props.order.id)}>&times;</a>
      </li>
    );
  }
}

Order.propTypes = {
  order: PropTypes.object,
  onDelete: PropTypes.func
}

export default Order;
