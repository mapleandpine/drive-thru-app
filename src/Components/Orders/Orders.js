import React, { Component } from 'react';
import PropTypes from 'prop-types';
import OrderItem from "./OrderItem";
import './Orders.css';

class Orders extends Component {

  constructor(props) {
    super(props);
    this.state = {
      newOrder:{}
    }
  }

  componentWillUpdate(nextProps, nextState) {
    console.log(nextProps, nextState);
  }

  newOrder(e) {
    this.setState({newOrder: new Order()}, function() {
      this.props.addOrder(this.state.newOrder);
    })
    e.preventDefault();
  }

  deleteOrder(order) {
    this.props.onDelete(order);
  }

  render() {
    let orderItems;
    if(this.props.orders) {
      orderItems = this.props.orders.map(order => {
        return (
          <OrderItem onDelete={this.deleteOrder.bind(this)} key={order.title} order={order} />
        )
      });
    }
    return (
      <div className="Orders">
        <h2>Orders</h2>
        {orderItems}
        <form onSubmit={this.newOrder.bind(this)}>
          <input type="submit" value="Submit"/>
        </form>
      </div>
    );
  }
}

Orders.propTypes = {
  orders: PropTypes.array,
  onDelete: PropTypes.func
}

export default Orders;
