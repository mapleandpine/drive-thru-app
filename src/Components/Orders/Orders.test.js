import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Orders from './Orders';

describe(Orders, () => {
  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <Orders />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
