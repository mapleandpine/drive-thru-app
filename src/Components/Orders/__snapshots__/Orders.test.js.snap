// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class Orders extends _react.Component {

  constructor(props) {
    super(props);
    this.state = {};
  }

  activeOrder(id) {
    this.props.onActiveOrder(id);
  }

  completeOrder(id) {
    this.props.onCompleteOrder(id);
  }

  deleteOrder(id) {
    this.props.onDeleteOrder(id);
  }

  render() {
    let orderItem;
    if (this.props.orders) {
      orderItem = this.props.orders.map(order => {
        const closed = order.closed;
        if (closed) {
          return (
            _react2.default.createElement(_OrderClosed2.default, { onActiveOrder: this.activeOrder.bind(this), onDeleteOrder: this.deleteOrder.bind(this), key: order.id, order: order, __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this }));

        } else {
          return (
            _react2.default.createElement(_Order2.default, { onActiveOrder: this.activeOrder.bind(this), onCompleteOrder: this.completeOrder.bind(this), onDeleteOrder: this.deleteOrder.bind(this), key: order.id, order: order, __source: { fileName: _jsxFileName, lineNumber: 37 }, __self: this }));

        }
      });
    }
    return (
      _react2.default.createElement('div', { className: 'Orders', __source: { fileName: _jsxFileName, lineNumber: 43 }, __self: this },
        _react2.default.createElement('h2', { className: 'heading', __source: { fileName: _jsxFileName, lineNumber: 44 }, __self: this }, 'Orders'),
        orderItem));


  }} renders and matches our snapshot 1`] = `
<div
  className="Orders"
>
  <h2
    className="heading"
  >
    Orders
  </h2>
</div>
`;
